newportxps/XPS_C8_drivers.py

```diff
@@ -58,20 +89,20 @@ def wrapper(*args, **kw):
     @withValidSocket
     def __sendAndReceive (self, socketId, command):
         # print("SEND REC ", command, type(command))
+        suffix = ',EndOfAPI'
         try:
             XPS.__sockets[socketId].send(str2bytes(command))
             ret = bytes2str(XPS.__sockets[socketId].recv(1024))
-            while (ret.find(',EndOfAPI') == -1):
+            while (ret.find(suffix) == -1):
                 ret += bytes2str(XPS.__sockets[socketId].recv(1024))
         except socket.timeout:
-            return [-2, '']
+            return -2, ''
         except socket.error as err: #  (errNb, errString):
             print( 'Socket error : ', err.errno, err)
-            return [-2, '']
+            return -2, ''
 
-        for i in range(len(ret)):
-            if (ret[i] == ','):
-                return [int(ret[0:i]), ret[i+1:-9]]
+        error, rest = ret[:-len(suffix)].split(',', 1)
+        return int(error), rest
```

Comment:
sorry there was one unrelated cleanup in this function - I updated it to return a tuple instead of a list (easier for the type checker to understand when the first and second item have different types), and I also eliminated this loop so that there was no possible path of returning `None` (also for the type checker)